### PR TITLE
feat(LT-5671): Axle: add new socket method to remove on behalf state

### PR DESCRIPTION
**Pull request by lykke-vashetsin**

### CODE REVIEW

File: src/Axle/Hubs/SessionHub.cs [hunk 1]
```diff
@@ -91,6 +91,18 @@ public Task<OnBehalfChangeResponse> SetOnBehalfAccount(string accountId)
             return sessionService.UpdateOnBehalfState(sessionId, accountId);
         }
 
+        public Task<bool> SetOffBehalfAccount(string accountId)
```
Comment (tarurar) [inline src/Axle/Hubs/SessionHub.cs:94]:
```suggestion
        public Task<bool> SetOffOnBehalfAccount(string accountId)
```

File: src/Axle/Services/SessionService.cs [hunk 2]
```diff
@@ -153,6 +153,40 @@ public async Task<OnBehalfChangeResponse> UpdateOnBehalfState(int sessionId, str
...
+                {
+                    var onBehalfOwner = await accountsService.GetAccountOwnerUserName(onBehalfAccount);
+
+                    if (string.IsNullOrEmpty(onBehalfOwner))
+                    {
+                        return false;
+                    }
+                }
+
+                await sessionRepository.Remove(sessionId, session.UserName, session.AccountId);
```
Comment (tarurar) [inline src/Axle/Services/SessionService.cs:179]: Don't you need to cover with lock only data modification (remove method)?

**Review by tarurar — CHANGES_REQUESTED** [review]

File: src/Axle/Hubs/SessionHub.cs (hunk 1 above)
Comment (lykke-vashetsin) [reply src/Axle/Hubs/SessionHub.cs]: done

File: src/Axle/Services/SessionService.cs (hunk 2 above)
Comment (lykke-vashetsin) [reply src/Axle/Services/SessionService.cs]: removed lock completely

**Review by tarurar — APPROVED** [review]

**Pull request merged**